### PR TITLE
Car by @AlexU238

### PR DESCRIPTION
**Pull request by AlexU238**

### CODE REVIEW

File: Car.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        this.mark = mark;
+        this.consumption = cons;
+        this.volume = vol;
+        this.mileage = mil;
+    }
+
+
+    public void drive(int km) {
+
+        double volumeForTheTrip = consumption * km;
```
Comment (malvitvik) [inline Car.java:21]: If you multiply two integers (without dot) - you're getting floating variable (with dot), right? Or it should be int/long type. 
P.S. 'consumption' could be a float (like 7.9) but you're used int.

File: Car.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        double fuelLeft = volume - volumeForTheTrip;
+
+
+        if (fuelLeft > 0) {
+            System.out.println("Проехала: " + km + " км\n");
+            System.out.println("Пробег: " + (km + mileage) + "км\n");
+            mileage = mileage + km;
+            volume = volume - volumeForTheTrip;
```
Comment (malvitvik) [inline Car.java:29]: The right part is the same as in the row 22.
You can use 'var1 = var2;'

File: Car.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    public void drive(int km) {
+
+        double volumeForTheTrip = consumption * km;
+        double fuelLeft = volume - volumeForTheTrip;
+
+
+        if (fuelLeft > 0) {
+            System.out.println("Проехала: " + km + " км\n");
+            System.out.println("Пробег: " + (km + mileage) + "км\n");
```
Comment (malvitvik) [inline Car.java:27]: You do 'mileage + km' action twice. It can easy be done once and displayed after that.

File: Car.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            mileage = mileage + km;
+            volume = volume - volumeForTheTrip;
+        } else if (fuelLeft <= 0) {
+            double distance = volume / consumption;
+            System.out.println("Проехала: " + distance + " км\n");
+            System.out.println("Пробег: " + (distance + mileage) + "км\n");
```
Comment (malvitvik) [inline Car.java:33]: You do 'distance + mileage' action twice. It can easy be done once and displayed after that.

File: Car.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            mileage = mileage + distance;
+            volume = 0;
+        } else if (consumption > volume || volume < volumeForTheTrip || volume == 0) {
+            System.out.println("Не хватает топлива. Требуется дозаправка\n");
+        }
+    }
+
+    public void reFuel(int fuelVolume) {
+        volume = volume + fuelVolume;
```
Comment (malvitvik) [inline Car.java:42]: A car with endless tank) I cab call 'car.reFuel(1000);' several times and get 
Заправлено на 1000 литров топлива
Топлива в баке: 100000 литров

File: Car.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            System.out.println("Пробег: " + (km + mileage) + "км\n");
+            mileage = mileage + km;
+            volume = volume - volumeForTheTrip;
+        } else if (fuelLeft <= 0) {
+            double distance = volume / consumption;
+            System.out.println("Проехала: " + distance + " км\n");
+            System.out.println("Пробег: " + (distance + mileage) + "км\n");
+            mileage = mileage + distance;
+            volume = 0;
+        } else if (consumption > volume || volume < volumeForTheTrip || volume == 0) {
```
Comment (malvitvik) [inline Car.java:36]: volumeForTheTrip = consumption * km;
fuelLeft = volume - volumeForTheTrip;

If _(consumption > volume)_ => _(volumeForTheTrip  > volume)_ automatically (or non-positive value is set to km) .
Then **fuelLeft** will be negative and _else if (fuelLeft <= 0)_

**Review by oleksandr-kazimirov — APPROVED** [review]

**Pull request closed without merging**